crates/ty_python_semantic/src/semantic_index/definition.rs

```diff
@@ -769,13 +769,14 @@ impl DefinitionKind<'_> {
                 target_range.cover(value_range)
             }
             DefinitionKind::AnnotatedAssignment(assign) => {
-                let target_range = assign.target.node(module).range();
+                let mut full_range = assign.target.node(module).range();
+                full_range = full_range.cover(assign.annotation.node(module).range());
```

Comment:
Due to how the syntax is structured, we could avoid this step if there is a right hand side. But this method is not performance critical, and I find it easier to reason about in this form: just always include the annotation.